include/modules/meta/base.inl

```diff
@@ -133,7 +133,7 @@ namespace modules {
     bool no_tag_built{true};
     bool fake_no_tag_built{false};
     bool tag_built{false};
-    auto mingap = std::max(1_z, format->spacing);
+    auto mingap = std::max(1.f, format->spacing.value);
```

Comment:
I think this is the only area of concern. This mingap is used down below to determine how many whitespaces to remove if a tag wasn't built.

This creates problems when the spacing isn't actually whitespace but an offset.

I would solve this by just deleting this variable and down below fix the logic that makes sure that spacing is only added before a tag, if the tag was actually built.

The function could look like this:

```cpp
  template <typename Impl>
  string module<Impl>::get_output() {
    std::lock_guard<std::mutex> guard(m_buildlock);
    auto format_name = CONST_MOD(Impl).get_format();
    auto format = m_formatter->get(format_name);
    bool no_tag_built{true};
    auto tag_builder = make_unique<builder>(m_bar);
    size_t start, end;
    string value{format->value};
    while ((start = value.find('<')) != string::npos && (end = value.find('>', start)) != string::npos) {
      if (start > 0) {
        if (no_tag_built) {
          // If no module tag has been built we do not want to add
          // whitespace defined between the format tags, but we do still
          // want to output other non-tag content
          auto trimmed = string_util::ltrim(value.substr(0, start), ' ');
          if (!trimmed.empty()) {
            m_builder->node(move(trimmed));
          }
        } else {
          m_builder->node(value.substr(0, start));
        }
        value.erase(0, start);
        end -= start;
        start = 0;
      }
      string tag{value.substr(start, end + 1)};
      if (tag.empty()) {
        continue;
      } else if (tag[0] == '<' && tag[tag.size() - 1] == '>') {
        if (CONST_MOD(Impl).build(tag_builder.get(), tag)) {
          m_builder->space(format->spacing);
          m_builder->append(tag_builder->flush());
          no_tag_built = false;
        }
        else {
          tag_builder->flush();
        }
      }
      value.erase(0, tag.size());
    }

    if (!value.empty()) {
      m_builder->append(value);
    }

    return format->decorate(&*m_builder, m_builder->flush());
  }
```

This isn't the best solution, but the best one where we don't need to massively modify the builder.

Now we can also completely remove the `remove_trailing_space` methods from the builder.